File: python/Pc.py

# import file kelas Processor, Disk dan Ram untuk digunakan composition
from Processor import Processor
from Disk import Disk
from Ram import Ram

class Pc():

    # deklarasi dan inisialisasi atribut private termasuk composition dengan objek Processor, Disk, dan Ram
    __Pobj = Processor
    __Dobj = Disk
    __Robj = Ram
    __totalPrice = 0


    # constructor tanpa parameter
    def __init__(self):
        self.__Pobj = Processor
        self.__Dobj = Disk
        self.__Robj = Ram
        self.__totalPrice = 0

    # constructor dengan parameter
    def __init__(self, inputP = Processor, inputD = Disk, inputR = Ram):
        self.__Pobj = inputP
        self.__Dobj = inputD
        self.__Robj = inputR


    # deklarasi setter sebagai method (Write)
    def setP(self, inputP = Processor):
        self.__Pobj = inputP

    def setD(self, inputD = Disk):
        self.__Dobj = inputD
    
    def setR(self, inputR = Ram):
        self.__Robj = inputR
    
    # setter pada total price adalah penjumlahan terhadap ketiga harga pada kelas spek PC
    def setTotalPrice(self, priceP = 0, priceD = 0, priceR = 0):
        self.__totalPrice =  float(priceP) + float(priceD) + float(priceR)



    # deklarasi getter sebagai method (Read Only)
    def getP(self):
        return self.__Pobj
    
    def getD(self):
        return self.__Dobj
    
    def getR(self):
        return self.__Robj
    
    def getTotal(self):
        return self.__totalPrice


    # deklarasi method Display untuk menampilkan data menggunakan method Getter
    def countTotal(self):
        print("Total Price    = $" + str(self.getTotal()))   

File: c++/Pc.cpp
#include <iostream>
#include <string>

using namespace std;

// import file yang akan composition pada kelas PC
#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

// deklarasi kelas Pc 
class Pc {
    
    // deklarasi dan inisialisasi atribut private termasuk composition dengan objek Processor, Disk, dan Ram
    private:
        Processor Pobj;
        Disk Dobj;
        Ram Robj;
        int totalPrice = 0;

    public:

        // constructor tanpa parameter
        Pc(){
            
        }

        // constructor dengan parameter kelas dan atribut
        Pc(Processor inputP, Disk inputD, Ram inputR){
            this->Pobj = inputP;
            this->Dobj = inputD;
            this->Robj = inputR;
        }

        
        // deklarasi setter sebagai method (Write)
        void setP(Processor P){
            this->Pobj = P;
        }

        void setD(Disk D){
            this->Dobj = D;
        }

        void setR(Ram R){
            this->Robj = R;
        }

        // setter pada total price adalah penjumlahan terhadap ketiga harga pada kelas spek PC
        void setTotalPrice(int priceP, int priceD, int priceR){
            this->totalPrice = priceP + priceD + priceR;
        }


        // deklarasi getter sebagai method (Read Only)
        Processor getP(){
            return this->Pobj;
        }

        Disk getD(){
            return this->Dobj;
        }

        Ram getR(){
            return this->Robj;
        }

        int getTotal(){
            return this->totalPrice;
        }

        // deklarasi method countTotal untuk menampilkan hasil penjumlahan harga dengan menggunakan method Getter
        void countTotal(){
            cout << "Total Price    = $" << this->getTotal() << endl;
        }

        // destructor
        ~Pc(){

        }

};